src/Buffer.h

```diff
@@ -107,7 +130,10 @@ class Buffer {
                               std::is_void<T2>::value,
                           "type mismatch constructing Buffer");
         } else {
-            Runtime::Buffer<T>::assert_can_convert_from(*(other.get()));
+            user_assert(Runtime::Buffer<T>::can_convert_from(*(other.get())))
```

Comment:
Worth adding a comment here about why we don't delegate to assert_can_convert_from()

Reply:
Done.